### ISSUE TITLE
Enabling a Category on Store View level does not set it's URL rewrite entry

### ISSUE THREAD

**Issue by amenk** [magento/magento2#28197]
We are creating categories via REST API. Now we discovered that the URL rewrite entry is not created for a second store, when setting the category name via REST PUT call

### Preconditions (*)
1. `n98-magerun2 install` - choose Magento 2.3.3
2. `./bin/magento sampledata:deploy`
3. `./bin/magento setup:upgrade`
4. Visit the admin panel and create a new website, store and store view using the same root category as the default store.
Store code = "austria"

### Steps to reproduce (*)
Call this script:

	BEARER=`n98-magerun2 admin:token:create admin`
	BASEURL="http://put-your-magento-base-url-here"

	echo '{"category":{"name":"Test","is_active":1,"parentId":2}}' > /tmp/datacreate.json

	curl -k -X POST $BASEURL/rest/V1/categories -H 'Content-Type: application/json' -H "Authorization: Bearer $BEARER" -d @/tmp/datacreate.json > /tmp/result.json

	ID=`jq .id /tmp/result.json`

	echo "New category ID is $ID"

	echo '{"category":{"name":"Test in austrian"}}' > /tmp/data_storeview.json
	curl -k -X PUT $BASEURL/rest/austria/V1/categories/$ID -H 'Content-Type: application/json' -H "Authorization: Bearer $BEARER" -d @/tmp/data_storeview.json

	echo "Now checking the URL rewrites -- is it there for all stores? (there should be two entries)"

	n98-magerun2 db:query "SELECT * FROM url_rewrite WHERE entity_id = $ID AND entity_type = 'category'";

### Expected result (*)
1. There should be two entries in the url_rewrite table, one for each store view

### Actual result (*)

	$ n98-magerun2 db:query "SELECT * FROM url_rewrite WHERE entity_id = 44 AND entity_type = 'category'";
	url_rewrite_id  entity_type     entity_id       request_path    target_path     redirect_type   store_id        description     is_autogenerated        metadata
	1422    category        44      test.html       catalog/category/view/id/44     0       1       NULL    1       NULL

Only one rewrite was created

### Additional / Workaround

When editing the URL key via Admin panel, we get

	$ n98-magerun2 db:query "SELECT * FROM url_rewrite WHERE entity_id = 44 AND entity_type = 'category'";
	url_rewrite_id  entity_type     entity_id       request_path    target_path     redirect_type   store_id        description     is_autogenerated        metadata
	1422    category        44      test.html       catalog/category/view/id/44     0       1       NULL    1       NULL
	1423    category        44      testfoo.html    catalog/category/view/id/44     0       2       NULL    1       NULL

**Comment by amenk** [comment]
I can reproduce it on a vanial 2.3.3 instance

**Comment by amenk** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Magento instance deployment temporary unavailable due to service maintenance.

**Comment by amenk** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Magento instance deployment temporary unavailable due to service maintenance.

**Comment by amenk** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @amenk. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @amenk, here is your [Magento instance](https://i-28197-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-28197-2-4-develop.instances.magento-community.engineering/admin_4875](https://i-28197-2-4-develop.instances.magento-community.engineering/admin_4875)
Login: `a4af1103` Password: `223e4d0d4bd1`
Instance will be terminated in up to 3 hours.

**Comment by amenk** [comment]
Reproduced in develop-2.4
	$ BEARER="91wi3rzds1wdlr75iohpso9v55v7d94o"
	$ BASEURL="https://i-28197-2-4-develop.instances.magento-community.engineering"
	$ echo '{"category":{"name":"Test","is_active":1,"parentId":2}}' > /tmp/datacreate.json
	$ curl -k -X POST $BASEURL/rest/V1/categories -H 'Content-Type: application/json' -H "Authorization: Bearer $BEARER" -d @/tmp/datacreate.json > /tmp/result.json
	$ ID=`jq .id /tmp/result.json`
	$ echo "New category ID is $ID"
	New category ID is 34
	$ echo '{"category":{"name":"Test in austrian"}}' > /tmp/data_storeview.json
	$ curl -k -X PUT $BASEURL/rest/austria/V1/categories/$ID -H 'Content-Type: application/json' -H "Authorization: Bearer $BEARER" -d @/tmp/data_storeview.json
	{"id":34,"parent_id":2,"name":"Test in austrian","is_active":true,"position":3,"level":2,"children":"","created_at":"2020-05-27 12:27:26","updated_at":"2020-05-27 12:27:57","path":"1\/2\/34","available_sort_by":[],"include_in_menu":true,"custom_attributes":[{"attribute_code":"is_anchor","value":"1"},{"attribute_code":"path","value":"1\/2\/34"},{"attribute_code":"children_count","value":"0"},{"attribute_code":"url_key","value":"test"},{"attribute_code":"url_path","value":"test"}]}
	$ echo "Now checking the URL rewrites -- is it there for all stores? (there should be two entries)"
	Now checking the URL rewrites -- is it there for all stores? (there should be two entries)

![image](https://user-images.githubusercontent.com/1087128/83019079-a1aea480-a026-11ea-885f-971e47a2197e.png)


There should be a URL rewrite for austria, but it's not

![image](https://user-images.githubusercontent.com/1087128/83018941-6c09bb80-a026-11ea-8245-50c94249a445.png)

**Comment by DanieliMi** [comment]
As workaround we tried to pass `save_rewrites_history` without success.  
Example payload:
```
{
  "category": {
    "name": "Test in austrian with save_rewrites_history",
    "custom_attributes": [
      {
        "attribute_code": "save_rewrites_history",
        "value": 1
      }
    ]
  }
}
```

**Comment by amenk** [comment]
The issue is a different one than originally reported.

Workaround here is: When supplying custom attribute url_key, the rewrite entry is generated.
This is consistent because via backend we also have to edit the URL key and not only the name.

The real problem is, that when we create the category, the URL rewrite table entry is only generated for the first store view.

**Comment by amenk** [comment]
closed in favor of #28395
